src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Controllers/FhirControllerTests.cs

```diff
@@ -0,0 +1,99 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using Hl7.Fhir.Model;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Core.Features.Context;
+using Microsoft.Health.Fhir.Api.Configs;
+using Microsoft.Health.Fhir.Api.Controllers;
+using Microsoft.Health.Fhir.Api.Features.ActionResults;
+using Microsoft.Health.Fhir.Core.Extensions;
+using Microsoft.Health.Fhir.Core.Features.Context;
+using Microsoft.Health.Fhir.Core.Features.Routing;
+using Microsoft.Health.Fhir.Core.Messages.Everything;
+using Microsoft.Health.Fhir.Core.Models;
+using NSubstitute;
+using Xunit;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Health.Fhir.Api.UnitTests.Controllers
+{
+    public class FhirControllerTests
+    {
+        private readonly FhirController _fhirController;
+        private readonly IMediator _mediator = Substitute.For<IMediator>();
+        private readonly ILogger<FhirController> _logger = Substitute.For<ILogger<FhirController>>();
+        private readonly RequestContextAccessor<IFhirRequestContext> _fhirRequestContextAccessor = Substitute.For<RequestContextAccessor<IFhirRequestContext>>();
+        private readonly IUrlResolver _urlResolver = Substitute.For<IUrlResolver>();
+        private readonly IAuthorizationService _authorizationService = Substitute.For<IAuthorizationService>();
+        private readonly HttpContext _httpContext = new DefaultHttpContext();
+
+        public FhirControllerTests()
+        {
+            _fhirController = GetController();
+            var controllerContext = new ControllerContext { HttpContext = _httpContext };
+            _fhirController.ControllerContext = controllerContext;
+        }
+
+        [Fact]
+        public async Task GivenAnEverythingOperationRequest_WhenValid_ThenProperResponseShouldBeReturned()
```

Comment:
>GivenAnEverythingOperationRequest_WhenValid_ThenProperResponseShouldBeReturned [](http://example.com/codeflow?start=26&length=78)

Can we have negative test(s)?
negative count, type contains garbage or non existing type, garbage continuation token?

Reply:
Sure! We have negative tests in E2E tests that include all valid/all invalid/some invalid type and garbage continuation token. As count is not supported, we do not verify the input `_count` parameter. I've looked into `FhirController.PatientEverythingById` and it does not do any checks, so I do not add it in unit tests and just keep those tests in E2E tests.